corona-warn-app/cwa-app-android
# Handle App Config uniformly with caching (EXPOSUREAPP-2747)

## Description
Our appconfig handling should meet the desired specs analog to iOS.
Most of the resiliance features were already built in but we didn't fall back to the last valid config all cases, now we do.
+ Some refactoring and better test coverage.

## How to test
* Run the app (fresh install too), check log for "AppConfig"
* Use the app in adverse conditions (without internet etc.)
* Check code and unit tests, tech spec should be met (see JIRA ticket)
* The only case where you should see errors are on a fresh install without internet where no valid config has ever been loaded
